R/link-href.R

```diff
@@ -29,6 +29,12 @@ href_expr <- function(expr, bare_symbol = FALSE) {
     if (grepl("^%.*%$", fun_name))
       return(NA_character_)
 
+    # prevent spurious links to Authors@R, content-home.html,
+    # toc: depth, etc.
+    # Black list `@`, `:`, and arithmetic operators
+    if (fun_name %in% c("@", "+", "-", "/", "*", "=", ":"))
```

Comment:
I think you could just make a single comment for this block and the block above: `# Exclude infix functions`.  Or maybe extract out in `is_infix()` function, and then we wouldn't even need the comment.

Reply:
Actually we should assert positively that it's a prefix function, also excluding the special forms:

```R
is_prefix <- function(fun) {
  if (grepl("^%.*%$", fun_name)) {
    return(FALSE)
  }
  
  infix <- c(
    "::", ":::", "$", "@", "[", "[[", "^", "-", "+", ":", "<",
    ">", "<=", ">=", "==", "!=", "!", "&", "&&", "|", "||", "~",
    "->", "->>", "<-", "<<-", "=", "?"
  )
  if (fun %in% infix) {
    return(FALSE)
  }
  
  special <- c(
    "(", "{", "if", "for", "while", "repeat", "next", "break", "function"
  )
  if (fun %in% infix) {
    return(FALSE)
  }
  
  TRUE
}
```